### ISSUE TITLE
`range()` documentation appears to be incorrect

### ISSUE THREAD

**Issue by pangolingo** [supabase/supabase#15222]
# Improve documentation

## Link 

[Limit the query to a range](https://supabase.com/docs/reference/javascript/range)

## Describe the problem

The docs state that "range" takes a starting index and ending index and returns the values between them. This implies that it's using the primary ID. But it actually works more like limit + offset.

## Describe the improvement

Consider something like this instead:

> Limit the query result by starting at an offset (`from`) and ending at the offset + `to`). Only records within this range are returned.
This respects the query order. The `from` and `to` values are 0-based and inclusive: from 1 to 3 will include the second, third and fourth rows of the query.

## Additional context

See this confused comment: https://github.com/orgs/supabase/discussions/1223#discussioncomment-3321739

**Comment by GaryAustin1** [comment]
I think the docs are further lacking that you really need to have an order clause and column to use range effectively.   There is no natural order (even primary key) to data returned in a query without and .order() modifier.   This is especially obvious with pagination as you can get duplicate rows between queries without an order column.

**Comment by pangolingo** [comment]
> I think the docs are further lacking that you really need to have an order clause and column to use range effectively. There is no natural order (even primary key) to data returned in a query without and .order() modifier. This is especially obvious with pagination as you can get duplicate rows between queries without an order column.

Good point. The [postgres docs](https://www.postgresql.org/docs/current/queries-limit.html) mention this:

> When using LIMIT, it is important to use an ORDER BY clause that constrains the result rows into a unique order. Otherwise you will get an unpredictable subset of the query's rows. You might be asking for the tenth through twentieth rows, but tenth through twentieth in what ordering? The ordering is unknown, unless you specified ORDER BY.

An updated suggestion:

> Limit the query result by starting at an offset (`from`) and ending at the offset `from + to`). Only records within this range are returned.
> This respects the query order and if there is no `order` clause the range could behave unexpectedly. The `from` and `to` values are 0-based and inclusive: from 1 to 3 will include the second, third and fourth rows of the query.

**Comment by gregnr** [comment]
Really great feedback @pangolingo 🎉 

Would you be interested in opening a small PR for this? The description comes from [supabase/postgrest-js](https://github.com/supabase/postgrest-js) repo:

https://github.com/supabase/postgrest-js/blob/aeb2ce9f1bdca64d0a0d12812c777b0e76df24ae/src/PostgrestTransformBuilder.ts#L103-L119

**Comment by pangolingo** [comment]
I've opened a PR on the postgrest-js` repo.
